src/component/VideoDetails.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
+class VideoDetail extends React.Component {
+
+
+   addToHistory = () =>{
+      let dummyArray = [];
+      dummyArray = JSON.parse(localStorage.getItem('user'));
+      dummyArray.history.push(this.props.video.id.videoId);
+      localStorage.setItem('user',JSON.stringify(dummyArray));
+   }
+
+
+   componentWillUpdate(){
+      if(!this.props.video)
+      {
+         return <div></div>
+      }
+         this.addToHistory();
+   }
+   componentDidMount(){
+
+      if(!this.props.video)
+      {
+         return <div></div>
+      }
+         this.addToHistory();
+   }
+
+   render() {
+      if (!this.props.video) {
+         return <div></div>
+      }
+
+      const {snippet,id} = this.props.video
+
+      const videoSrc = `https://www.youtube.com/embed/${id.videoId}?autoplay=1`
```

Comment:
@RajanTank What if the `id` is not present? 

Rule - `Validate your inputs when you receive them, before you move ahead and do further processing`